frontend/src/metabase/core/components/TabLink/TabLink.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { MouseEvent, useCallback, useContext } from "react";
+import { LinkProps } from "metabase/core/components/Link";
+import { getTabId, getTabPanelId, TabContext } from "../Tab";
+import { TabLinkRoot, TabLabel } from "./TabLink.styled";
+
+export interface TabLinkProps<T> extends LinkProps {
+  value?: T;
```

Comment:
I'm not sure what are you trying to achieve with generic, but from the usage in this PR, seems like it is always string, so I wonder if this could be just `string` so it's simple?

Reply:
I'm just mirroring the interface of [Tab](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/core/components/Tab/Tab.tsx) from the same group of components. `TabLink` is meant to be an alternative to `Tab`; think the more similar API they have the better. WDYT?

Reply:
Agree that the more similar API the better. But I still think generic here doesn't convey much. Think about using tabs, do we actually want to be able to pass anything as a value like objects, lists, maps? It'd make more sense to say maybe this can either be a `number` or a `string` ([I found the number being used here.](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/core/components/TabList/TabList.stories.tsx#L26)).

Reply:
Sounds good, I'd better do it in another PR though not to blow up the scope of this one too muh